### ISSUE TITLE
IJulia prints password

### ISSUE THREAD

**Issue by andreasnoack** [JuliaLang/IJulia.jl#678]
I just updated my packages in a notebook and was asked for my password. It was printed in clear text twice which is probably not a good idea. Below is an example where I've substituted my token with `[MY SECRET TOKEN]`
```julia
INFO: Updating HttpParser master...
INFO: Updating TableView master... 1f29c98f → 4c7556b2
STDIN> andreasnoack
Username for 'https://github.com' [andreasnoack]:
Password for 'https://andreasnoack@github.com':
STDIN> [MY SECRET TOKEN]
Password for 'https://[MY SECRET TOKEN]@github.com':INFO: Updating IRTM master... 82dd8d34 → 1ca6d3ec
INFO: Updating SharedDataStructures master...
INFO: Updating RemoteMonitor master...
```

**Comment by stevengj** [comment]
There is a `IJulia.readprompt(prompt, password=true)` method that could be called here.  It just needs to be hooked up to the correct Base API.

One problem here is that `Base.prompt` and `Base.getpass` don't accept a `stdin` parameter, making them difficult to override, though I suppose we could force it with `eval`.  Since these are internal routines, however, maybe they could be patched for 1.0.

**Comment by stevengj** [comment]
Just posted a PR for Julia that will allow us to fix this.
